assets/app/actions/siteActions.js

```diff
@@ -75,6 +76,23 @@ export default {
           });
         });
       });
-    })
+    });
+  },
+
+  fetchContent(site, path) {
+    function dispatchChildren(site, path, files) {
```

Comment:
I could probably go either way, but I like using fetch instead of dispatch here. I realize that the function is mostly dispatching an action but the action creator still calls the fetch method to make a network request and get the content, and that's why you'd want to call it in the first place.